contributed/ra_dec_dc2_run2.2i_dask.ipynb

```diff
@@ -0,0 +1,1455 @@
+    "    ra_dec = ra_dec * ddf_region\n",
+    "\n",
+    "    max_delta_ra = dc2_run2x_wfd_df['ra'][3] - dc2_run2x_wfd_df['ra'][2]\n",
+    "    delta_dec = dc2_run2x_wfd_df['dec'][1] - dc2_run2x_wfd_df['dec'][3]\n",
+    "    grow_buffer = 0.05\n",
+    "\n",
+    "    # Notice that these are specified in increasing RA left->right\n",
+    "    # We rely on the invert_xaxis True above to flip this in the display\n",
+    "    # It's important to get this right because these ranges are used for data selection\n",
+    "    # and then the range is flipped in the display.\n",
+    "    ra_dec.opts(xlim=(dc2_run2x_wfd_df['ra'][2] - max_delta_ra * grow_buffer,\n",
+    "                dc2_run2x_wfd_df['ra'][3] + max_delta_ra * grow_buffer))\n",
+    "    ra_dec.opts(ylim=(dc2_run2x_wfd_df['dec'][3] - delta_dec * grow_buffer,\n",
+    "                dc2_run2x_wfd_df['dec'][1] + delta_dec * grow_buffer))\n",
+    "\n",
+    "    return ra_dec"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_ra_dec(df, dc2_run2x_wfd_df=dc2_run2x_wfd_df, dc2_run2x_ddf_df=dc2_run2x_ddf_df,\n",
+    "                show_dc2_region=True, cmap=\"bmy\", bins=100, cmin=10):\n",
+    "    \"\"\"Show rasterized RA, Dec object density.\n",
+    "    \n",
+    "    We're just doing this on a rectilinear grid\n",
+    "    The distortion is noticeable from the lowest to highest Dec in the change in density due to the change in area.\"\"\"\n",
+    "    points_ra_dec = hv.Points(df, kdims=[hv.Dimension('ra', soft_range=(dc2_run2x_wfd[2][0], dc2_run2x_wfd[3][0])),\n",
+    "                                         hv.Dimension('dec', soft_range=(dc2_run2x_wfd[3][1], dc2_run2x_wfd[1][1]))])\n",
+    "    # We have to define the colormap here now, because the opts aren't passed through the datashade->Points.\n",
+    "    # See, e.g., https://github.com/holoviz/holoviews/issues/4125\n",
+    "    ra_dec = datashade(points_ra_dec, cmap=process_cmap(cmap, provider=\"colorcet\"), precompute=True)\n",
+    "    ra_dec = ra_dec.opts(invert_xaxis=True)  # Flip to East left\n",
+    "#    ra_dec = ra_dec.opts(precompute=True)\n",
+    "    if show_dc2_region:\n",
+    "        ra_dec = overlay_dc2_region(ra_dec, dc2_run2x_wfd_df=dc2_run2x_wfd_df, dc2_run2x_ddf_df=dc2_run2x_ddf_df)\n",
+    "   \n",
+    "    return ra_dec"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ra_dec = plot_ra_dec(ddf)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ra_dec.opts(width=800, height=700)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Save a PNG of the plot\n",
+    "hv.save(ra_dec, 'DC2_Run2.2i_DR6_ra_dec.png', fmt='png')"
```

Comment:
Just a note that this didn't work for me since I didn't have the "selenium package" installed.

Reply:
Yeah, this fails on NERSC.  It works fine for me locally.  But, in general, actually saving plots from HoloViews+Bokeh is frustratingly fragile.